lib/msal-core/src/utils/WindowUtils.ts

```diff
@@ -345,6 +348,7 @@ export class WindowUtils {
         // Office.js sets history.replaceState to null
         if (typeof history.replaceState === "function") {
             // Full removes "#" from url
+            window.location.replace(`${window.location.pathname}${window.location.search}`);
```

Comment:
I think this could be done before the `if` (i.e. for both when history.replaceState is available or not)

Reply:
Is `history.replaceState` still needed at all if we use `window.location.replace`?

Reply:
Yes, I tried multiple approaches and this is the only one that works. `window.location.replace` sets the url in the browser, but by then the previous URL is already in the history object, so the replaceState line is also necessary to remove it.

Reply:
So, after a lot of fiddling, `history.replaceState` will work, but the hash has to be set to an empty string beforehand so the id_token is properly cleared from the history entry. With the latest changes, both redirect and popup use the same `clearUrlFragment` method and they're both leaving a single history entry without the id_token nor the octothorpe at the end of the URL.